packages/web3-eth/src/rpc_method_wrappers.ts

```diff
@@ -988,7 +988,21 @@ export async function getLogs<ReturnFormat extends DataFormat>(
 	filter: Filter,
 	returnFormat: ReturnFormat,
 ) {
-	const response = await ethRpcMethods.getLogs(web3Context.requestManager, filter);
+	let { toBlock, fromBlock } = filter;
+	if (!isNullish(toBlock)) {
+		if (typeof toBlock === 'number' || typeof toBlock === 'bigint') {
+			toBlock = numberToHex(toBlock);
```

Comment:
lgtm, some of these lines are not covered in unit test,so could you add unit test ?